event/register_oracle_event.go

```diff
@@ -21,9 +23,32 @@ func (e RegisterOracleEvent) GetEventAttributeValue() string {
 	return "'RegisterOracle'"
 }
 
-func (e RegisterOracleEvent) EventHandler(event ctypes.ResultEvent) error {
-	// TODO: Verifying Remote Attestation and Executing Vote Txs
+func (e RegisterOracleEvent) EventHandler(event ctypes.ResultEvent, svc *service.Service) error {
+	addressValue := event.Events[types.EventTypeRegistrationVote+"."+types.AttributeKeyOracleAddress][0]
 
-	fmt.Println("RegisterOracle Event Handler")
+	oracleRegistration, err := svc.GrpcClient.GetOracleRegistration(addressValue, svc.UniqueID)
+	if err != nil {
+		return err
+	}
+
+	err = verifyRemoteReport(oracleRegistration)
+	if err != nil {
+		return err
+	}
+
+	// TODO: Executing Vote Txs
+	return nil
+}
+
+func verifyRemoteReport(oracleRegistration *types.OracleRegistration) error {
+	selfEnclaveInfo, err := sgx.GetSelfEnclaveInfo()
+	if err != nil {
+		return err
+	}
```

Comment:
You don't need to get self enclave info here.
Actually, `service.Service`'s `UniqueID` is from the self enclave info ([link](https://github.com/medibloc/panacea-doracle/blob/f4b0740670764a08bc97a85dc9d76ce66b154048/service/service.go#L39)).
So you can refactor the `service.Service` struct to have the whole enclave info, and utilize it.

Reply:
Fixed in [b127b4f](https://github.com/medibloc/panacea-doracle/pull/32/commits/b127b4f4df1c8c29eed5333bb4fe8d495a738cbb).